blocks/functions.js

```diff
@@ -0,0 +1,230 @@
+/**
+ * @license
+ * PXT Blockly fork
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/**
+ * @fileoverview Utility functions for handling functions (pxt-blockly's custom procedures).
+ * @author Microsoft MakeCode
+ */
+'use strict';
+
+/**
+ * Type to represent a function parameter
+ * @typedef {Object} FunctionParameter
+ * @property {string} id The blockly ID of the param
+ * @property {string} name the name of the param
+ * @property {string} type the type of the param (string, number, boolean or a custom type)
+ */
+
```

Comment:
nit: remove new line

Reply:
The JSDoc is not related to the object being defined below, it just declares a type so the other JSDoc in the file can use that type

I fixed by moving the Utils declaration above the JSDoc